worker/base/worker.go

```diff
@@ -269,3 +275,24 @@ func (w *Worker) Exporter(name string) (exporter.Exporter, error) {
 func (w *Worker) InstructionCache() instructioncache.InstructionCache {
 	return w.cache
 }
+
+// DetermineAnnotations returns ID and labels.
+// utility function, could be moved to the constructor logic?
+func DetermineAnnotations(daemonInstanceID, executor, snapshotter string) (string, map[string]string) {
+	hostname, err := os.Hostname()
+	if err != nil {
+		hostname = "unknown"
+	}
+	labels := map[string]string{
+		worker.LabelOS:          runtime.GOOS,
+		worker.LabelArch:        runtime.GOARCH,
+		worker.LabelExecutor:    executor,
+		worker.LabelSnapshotter: snapshotter,
+		worker.LabelHostname:    hostname,
+	}
+	if daemonInstanceID == "" {
+		daemonInstanceID = hostname
+	}
+	id := executor + "-" + snapshotter + "@" + daemonInstanceID
```

Comment:
Should we just use random string?

Reply:
Yes, I think ID should be unique. We can show these fields on the non-verbose output of the command.

Reply:
Store it in the state directory on first run.